### ISSUE TITLE
Sudo prompt timeout on redhat hosts only

### ISSUE THREAD

**Issue by Mark0016** [ansible/ansible#67969]
##### SUMMARY

Sudo prompt times out on RedHat based hosts only. With pipelining enabled everything works fine.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

sudo

##### ANSIBLE VERSION

```paste below
ansible 2.9.3
  config file = /home/mark/.ansible.cfg
  configured module search path = ['/home/mark/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.8 (default, Nov 21 2019, 19:31:34) [GCC 8.3.1 20190507 (Red Hat 8.3.1-4)]
```

##### CONFIGURATION

```paste below
DEFAULT_HOST_LIST(/home/mark/git/ansible-sudo-bug/ansible.cfg) = ['/home/mark/git/ansible-sudo-bug/inventory']
DEPRECATION_WARNINGS(/home/mark/git/ansible-sudo-bug/ansible.cfg) = False
HOST_KEY_CHECKING(/home/mark/git/ansible-sudo-bug/ansible.cfg) = False
```

##### OS / ENVIRONMENT

The environment I'm testing in consists of VMs all with different Linux distributions. Their name reflects the distribution I'ts running. Here's my inventory:
```
[testservers]
centos8
centos7
rhel8
fedora31
debian
ubuntu16-04
ubuntu18-04
```
The host `debian` is Debian 10 stable.
For testing a user was created named `sudoall` which has ssh-keys installed from the control host. The user has the following sudo policy (requires password):
```
sudoall    ALL=(ALL) ALL
```

##### STEPS TO REPRODUCE

Run this playbook with `-K` option to ask for become password.

```yaml
---
- name: 'Test sudo bug'
  gather_facts: no
  hosts: testservers
  user: sudoall
  become_method: sudo
  become_user: root
  tasks:
    - name: 'Run id'
      command: id
      become: yes
      register: id_result

    - name: 'Show id stdout'
      debug:
        var: id_result.stdout
```

##### EXPECTED RESULTS

Playbook should run `id` and print it's stdout which should show current user as root.

##### ACTUAL RESULTS

The expected happens on hosts `debian`,`ubuntu16-04` and `ubuntu18-04`. All other hosts (`centos8`, `centos7`,`rhel8`,`fedora31`) fail with `Timeout (12s) waiting for privilege escalation prompt: ` despite the prompt showing up in the stdout in the debug output like
```
11936 1583235011.48993: stdout chunk (state=0):
>>>[sudo via ansible, key=euxtwlkwnsbduyzuqhrndztatxspjtpq] password:<<<
```
If I enable pipelining the playbook executes an all hosts as expected. 

```paste below
PLAY [Test sudo bug] ********************************************************************************

TASK [Run id] ***************************************************************************************
[WARNING]: Platform linux on host debian is using the discovered Python interpreter at
/usr/bin/python, but future installation of another Python interpreter could change this. See
https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html for more
information.

changed: [debian]
changed: [ubuntu16-04]
changed: [ubuntu18-04]
fatal: [centos7]: FAILED! => {"msg": "Timeout (12s) waiting for privilege escalation prompt: "}
fatal: [fedora31]: FAILED! => {"msg": "Timeout (12s) waiting for privilege escalation prompt: "}
fatal: [rhel8]: FAILED! => {"msg": "Timeout (12s) waiting for privilege escalation prompt: "}
fatal: [centos8]: FAILED! => {"msg": "Timeout (12s) waiting for privilege escalation prompt: "}

TASK [Show id stdout] *******************************************************************************
ok: [ubuntu16-04] => {
    "id_result.stdout": "uid=0(root) gid=0(root) groups=0(root)"
}
ok: [debian] => {
    "id_result.stdout": "uid=0(root) gid=0(root) groups=0(root)"
}
ok: [ubuntu18-04] => {
    "id_result.stdout": "uid=0(root) gid=0(root) groups=0(root)"
}

PLAY RECAP ******************************************************************************************
centos7                    : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
centos8                    : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
debian                     : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
fedora31                   : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
rhel8                      : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
ubuntu16-04                : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
ubuntu18-04                : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```
**Very verbose and debug log** https://raw.githubusercontent.com/Mark0016/ansible-sudo-prompt-issue/master/test.log
**Same but limited to only centos8 host** https://raw.githubusercontent.com/Mark0016/ansible-sudo-prompt-issue/master/test_centos8.log

I've made a repo for all the files used for my testing with extra details and all the logs included as well here https://github.com/Mark0016/ansible-sudo-prompt-issue.
I can provide other files/configuratios as needed.

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/plugins/become/sudo.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/become/sudo.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Mark0016** [comment]
Looking through the logs again the following sequence shows up on stdout before the password prompt
```
>>>7[r[9999;9999H[6n<<<
```
The weird characters are `ESC` (`0x1b` or `0o33`).
This looks to me like a terminal escape sequence. It only shows up on the failed machines so it's possible this might be causing ansible to somehow not detect the prompt.

**Comment by Mark0016** [comment]
OK, My bad.
**This is not an issue with ansible**
Since I was accessing the VMs via the serial console I added a line in `/etc/profile.d` to run xterm-resize on very login. This escape sequence is produced by `resize` when it runs (figured out from tying to decode it). It appears on RedHat scripts in `/etc/profile.d` are ran even on non-login shells.
I added a condition to my script in `/etc/profile.d` to only run `resize` if the tty is /dev/ttyS0. This way the playbook executes correctly.

It still looks like ansible doesn't like terminal escape sequences on the stdout but if that's not something that needs to be fixed I understand.

I will close this issue.
